### PR TITLE
Use async fs methods

### LINKED ISSUE #147: Use async fs instead of fs.*Sync calls
[0x](https://github.com/davidmarkclements/0x), a cool tool by @davidmarkclements shows, that (for my build) `fs.realpathSync` calls block the running for 15% of the total time.

image

### PR DESCRIPTION
**Pull request by kaksmet**
Fixes #147

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,8 +13,12 @@ const ES6_BROWSER_EMPTY = '\0node-resolve:empty.js';
 // which deploy both ESM .mjs and CommonJS .js files as ESM.
 const DEFAULT_EXTS = [ '.mjs', '.js', '.json', '.node' ];
 
+const existsAsync = file => new Promise(fulfil => fs.exists(file, exists => fulfil(exists)));
```
Comment (lukastaegert) [inline src/index.js]: Minor point: This could be `...fs.exists(file, fulfil))`

File: src/index.js [hunk 2]
```diff
@@ -306,11 +310,15 @@ export default function nodeResolve ( options = {} ) {
 						browserMapCache.set(resolved, packageBrowserField);
 					}
 
+					if ( hasPackageEntry && !preserveSymlinks && resolved ) {
+						return existsAsync( resolved )
+							.then(exists => exists ? realpathAsync( resolved ) : resolved);
+					} else {
```
Comment (lukastaegert) [inline src/index.js]: No reason for else after return, but this is mostly a style question

File: src/index.js (hunk 1 above)
Comment (kaksmet) [reply src/index.js]: fixed

File: src/index.js (hunk 2 above)
Comment (kaksmet) [reply src/index.js]: fixed

**Comment by kaksmet** [comment]
Review comments have been addressed.

**Pull request merged**